diff --git a/src/runtime/Makefile b/src/runtime/Makefile
index 9fd4165..b78ecf6 100644
--- a/src/runtime/Makefile
+++ b/src/runtime/Makefile
@@ -1,7 +1,7 @@
 GIT_COMMIT := $(shell cat version)
 CC            = clang
 CFLAGS        = -std=gnu99 -Os -D_FILE_OFFSET_BITS=64 -DGIT_COMMIT=\"$(GIT_COMMIT)\" -T data_sections.ld -ffunction-sections -fdata-sections -Wl,--gc-sections -static -Wall -Werror -static-pie
-LIBS          = -lsquashfuse -lsquashfuse_ll -lzstd -lz -lfuse3 -lmimalloc
+LIBS          = -L /usr/local/lib/mimalloc-2.2 -lsquashfuse -lsquashfuse_ll -lzstd -lz -lfuse3 -lmimalloc -llzma -llz4 -llzo2
 
 all: runtime
 
diff --git a/src/runtime/fuseprivate.h b/src/runtime/fuseprivate.h
new file mode 100644
index 0000000..30ca909
--- /dev/null
+++ b/src/runtime/fuseprivate.h
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2014 Dave Vasilevsky <dave@vasilevsky.ca>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE AUTHOR(S) ``AS IS'' AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL THE AUTHOR(S) BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT
+ * NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+ * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+#ifndef SQFS_FUSEPRIVATE_H
+#define SQFS_FUSEPRIVATE_H
+
+#include <squashfuse/squashfuse.h>
+
+#include <fuse.h>
+
+#define NOTIFY_SUCCESS 's'
+#define NOTIFY_FAILURE 'f'
+
+/* Common functions for FUSE high- and low-level clients */
+
+/* Populate an xattr list. Return an errno value. */
+int sqfs_listxattr(sqfs *fs, sqfs_inode *inode, char *buf, size_t *size);
+
+/* Print a usage string */
+int sqfs_usage(char *progname, bool fuse_usage, bool ll_usage);
+
+/* Parse command-line arguments */
+typedef struct {
+	char *progname;
+	const char *image;
+	const char *subdir;
+	int mountpoint;
+	size_t offset;
+	unsigned int idle_timeout_secs;
+	int uid;
+	int gid;
+	const char *notify_pipe;
+} sqfs_opts;
+int sqfs_opt_proc(void *data, const char *arg, int key,
+	struct fuse_args *outargs);
+
+/* Get filesystem super block info */
+int sqfs_statfs(sqfs *sq, struct statvfs *st);
+void notify_mount_ready(const char *notify_pipe, char status);
+void notify_mount_ready_async(const char *notify_pipe, char status);
+
+#endif
diff --git a/src/runtime/runtime.c b/src/runtime/runtime.c
index 2ebb3c8..d6d2bb6 100644
--- a/src/runtime/runtime.c
+++ b/src/runtime/runtime.c
@@ -39,7 +39,7 @@
 #include <stddef.h>
 
 #include <squashfuse/ll.h>
-#include <squashfuse/fuseprivate.h>
+#include "fuseprivate.h"
 
 extern dev_t sqfs_makedev(int maj, int min);
 
